components/camel-debezium/src/main/java/org/apache/camel/component/debezium/DebeziumEndpoint.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.debezium.configuration.EmbeddedDebeziumConfiguration;
+import org.apache.camel.component.debezium.configuration.MySqlConnectorEmbeddedDebeziumConfiguration;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.support.DefaultEndpoint;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.source.SourceRecord;
+
+/**
+ * Represents a Debezium endpoint which is used for interacting with Debezium
+ * embedded engine.
+ */
+@UriEndpoint(firstVersion = "3.0.0", scheme = "debezium", title = "Debezium", syntax = "debezium:connector-name", label = "cdc", consumerOnly = true)
+public class DebeziumEndpoint extends DefaultEndpoint {
+
+    @UriParam
+    private EmbeddedDebeziumConfiguration configuration;
+
+    // is only meant to include different configurations for each connector that is generated by camel-maven plugin
+    @UriParam
+    private MySqlConnectorEmbeddedDebeziumConfiguration mySqlConnectorEmbeddedDebeziumConfiguration;
+
+    public DebeziumEndpoint(String uri, DebeziumComponent component, EmbeddedDebeziumConfiguration config) {
+        super(uri, component);
+        this.configuration = config;
+    }
+
+    public DebeziumEndpoint() {
+    }
+
+    @Override
+    public Producer createProducer() throws Exception {
+        throw new UnsupportedOperationException("Cannot produce from a DebeziumEndpoint: "
+                                                + getEndpointUri());
+    }
+
+    public Consumer createConsumer(Processor processor) throws Exception {
+        return new DebeziumConsumer(this, processor);
+    }
+
+    public boolean isSingleton() {
+        return true;
+    }
+
+    public EmbeddedDebeziumConfiguration getConfiguration() {
+        return configuration;
+    }
+
+    public void setConfiguration(EmbeddedDebeziumConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    public MySqlConnectorEmbeddedDebeziumConfiguration getMySqlConnectorEmbeddedDebeziumConfiguration() {
+        return mySqlConnectorEmbeddedDebeziumConfiguration;
+    }
+
+    public ExecutorService createExecutor() {
+        return getCamelContext().getExecutorServiceManager().newSingleThreadExecutor(this,
+                                                                                     "DebeziumConsumer");
+    }
+
+    public Exchange createDbzExchange(final SourceRecord record) {
+        final Exchange exchange = super.createExchange();
+
+        final Message message = exchange.getIn();
+
+        message.setHeader(DebeziumConstants.FIELD_IDENTIFIER, record.topic());
+        message.setHeader(DebeziumConstants.FIELD_KEY, record.key());
+        message.setHeader(DebeziumConstants.FIELD_SOURCE, record.sourceOffset());
```

Comment:
Why is it you expose the source offset here? Is it really needed? It's mostly specific to the actual Kafka transport.

Reply:
Correct me if I am wrong, but I think it also contains information about the binlog file and its position (in case of mysql, not sure about others), hence that is why I put it. Or maybe I am missing something here?

Reply:
I will remove it as replace it with the `source` metadata from the `payload|value`